src/de.hpi.swa.trufflesqueak/src/de/hpi/swa/trufflesqueak/util/UnsafeUtils.java

```diff
@@ -66,6 +66,50 @@ public static void copyShorts(final short[] src, final long srcPos, final short[
                         dest, Unsafe.ARRAY_SHORT_BASE_OFFSET + destPos * Unsafe.ARRAY_SHORT_INDEX_SCALE, Short.BYTES * length);
     }
 
+    public static long allocateNativeBytes(final byte[] src) {
+        final long address = UNSAFE.allocateMemory((long) src.length * Byte.BYTES);
+        UNSAFE.copyMemory(src, Unsafe.ARRAY_BYTE_BASE_OFFSET, null, address, src.length * Byte.BYTES);
+        return address;
+    }
+
+    public static long allocateNativeShorts(final short[] src) {
+        final long address = UNSAFE.allocateMemory((long) src.length * Short.BYTES);
+        UNSAFE.copyMemory(src, Unsafe.ARRAY_SHORT_BASE_OFFSET, null, address, (long) src.length * Short.BYTES);
```

Comment:
Here and in similar helpers:
```suggestion
        final long numBytes = src.length * Short.BYTES;
        final long address = UNSAFE.allocateMemory(numBytes);
        UNSAFE.copyMemory(src, Unsafe.ARRAY_SHORT_BASE_OFFSET, null, address, numBytes);
```